dart/dynamics/BodyNode.h

```diff
@@ -613,6 +591,60 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
```

Comment:
I see. The process of type deduction for templated function doesn't check if the passed-in parameter type can be implicitly converted to one of possible types. Good to know!